spec/search_paths_spec.rb

```diff
@@ -0,0 +1,117 @@
+require 'spec_helper'
+require 'ronin/wordlists/search_paths'
+require 'ronin/wordlists/exceptions'
+
+describe Ronin::Wordlists::SearchPaths do
+  subject { described_class.new([path1, path2]) }
+
+  let(:path1) { File.expand_path(File.join(__dir__, '..', 'spec', 'fixtures', 'wordlists')) }
+  let(:path2) { File.expand_path(File.join(__dir__, '..', 'spec', 'fixtures', 'extra_wordlists')) }
+
+  describe "#initialize" do
+    it "must initialize #paths and call #<< for each one" do
+      expect(subject.paths).to all(be_kind_of(Ronin::Wordlists::WordlistDir))
+      expect(subject.paths.map(&:path).sort).to eq([path2, path1])
+    end
+  end
+
+  describe ".[]" do
+    context "for single path" do
+      subject { described_class[path1] }
+
+      it "must initialize SearchPaths instance" do
+        expect(subject).to be_kind_of(described_class)
+        expect(subject.paths.map(&:path)).to eq([path1])
+      end
+    end
+
+    context "for multiple paths" do
+      subject { described_class[path1, path2] }
+
+      it "must initialize SearchPaths instance" do
+        expect(subject).to be_kind_of(described_class)
+        expect(subject.paths.map(&:path)).to eq([path2, path1])
+      end
+    end
+  end
+
+  describe "#each" do
+    let(:wordlist_dirs) { [Ronin::Wordlists::WordlistDir.new(path2), Ronin::Wordlists::WordlistDir.new(path1)] }
+
+    it "must iterate over #paths and yield each directory" do
+      yielded_values = []
+
+      subject.each do |value|
+        yielded_values << value
+      end
+
+      expect(yielded_values.size).to eq(2)
+      expect(yielded_values.map(&:path)).to eq(wordlist_dirs.map(&:path))
+    end
+  end
+
+  describe "#<<" do
+    before do
+      subject << path1
+    end
+
+    it "must add a new WordlistDir instance to search paths and return self" do
+      expect(subject).to be_kind_of(described_class)
```

Comment:
This line can be removed since we're not changing the value of `subject` in this spec, just calling `subject#<<` which changes the contents of `subject`.